### DISCUSSION TITLE
Incremental snapshots

### DISCUSSION THREAD

**Discussion by inatale** [kubernetes-sigs/azuredisk-csi-driver#1218] [Q&A] [answered]
Azure Disks CSI driver for Kubernetes supports incremental snapshots (VolumeSnapshotClass with incremental: "true"). But how could I create incremental snapshot? Do I need to create each time VolumeSnapshot objects with different names and any such object after 1st one will be incremental?

I tried to create 2 different VolumeSnapshot from the same PVC and after that I deleted the 1st one and I was able to restore from 2nd snapshot successfully. How is it possible if 2nd snapshot has only incremental data?

It's also not possible to view actual snapshots size using Azure portal because it displays overall allocated disk size.

**Comment by andyzhangx** [comment, marked as answer]
it's answered here: 
Now, if you deleted the first incremental snapshot the second and the third snapshots continue to function normally as incremental snapshots are independent of each other. The system merges the data occupied by the first snapshot with the second snapshot under the hood to ensure that the second and the third snapshots are not impacted due to the deletion of the first snapshot. The second snapshot now occupies 120 GiB of data. 

https://azure.microsoft.com/en-gb/blog/announcing-general-availability-of-incremental-snapshots-of-managed-disks/
